webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx

```diff
@@ -82,8 +82,8 @@ const useModalExperimentMove = ({ onClose }: Props): ModalHooks => {
   }, []);
 
   useEffect(() => {
-    ensureProjectsFetched(workspaceId);
-  }, [workspaceId, ensureProjectsFetched]);
+    fetchWorkspaceProjects(workspaceId);
```

Comment:
`useModalExperimentMove` is used by each row of the table. so when every usage kicks off its own request there end up being a lot, see below:

https://user-images.githubusercontent.com/28383080/214236833-425360fe-5d0d-4fc5-a1b5-8f3a9ce38ca5.mov

i think that doing this on master branch avoids that problem:
```if (context.projectsByIndex.get(`byworkspace-${workspaceId}`)) return;```

Reply:
`projectsByIndex` should really be private. Any particular reason you didn't maintain the `ensureX` function?